tests/Collector/SilverStripeCollectorTest.php

```diff
@@ -172,8 +171,8 @@ public function testGetWidgets()
     public function testGetAssets()
     {
         $expected = array(
-            'base_path' => '/debugbar/javascript',
-            'base_url' => 'debugbar/javascript',
+            'base_path' => '/resources/lekoala/silverstripe-debugbar/javascript',
```

Comment:
Don't hard-code this assumption; Use the resources API.

Reply:
A better test isn't an explicit hard-coded array result; It's that the path physically exists, and that the URL Is accessible.

Reply:
I agree - I think I should move away from relying on the module being in a particular place.

Travis testing is a nightmare with this though - because it runs the `composer install` from _inside_ the module, it means all the resources are in the wrong places. Will have a play again tomorrow - and see if I can figure out a standard for testing with modules that expose resources.

Reply:
I agree and disagree - a known input should result in a known output for a test. whilst we don't want to tightly couple our tests to these exact values, these assumptions should still hold...

Reply:
"installed in vendor or in root" isn't a known input it's a variable, and it shouldn't matter which it is.

```
$config = $this->collector->getAssets();
$this->assertDirectoryExists(BASE_PATH . '/' . $config['base_path]);
```

Otherwise, test for either of the two possible valid locations.